Derivs/src/derivs.hxx

```diff
@@ -332,19 +332,18 @@ inline CCTK_ATTRIBUTE_ALWAYS_INLINE
 template <int deriv_order, typename T>
 inline CCTK_ATTRIBUTE_ALWAYS_INLINE
     CCTK_DEVICE CCTK_HOST Arith::vec<Arith::simd<T>, Loop::dim>
-    calc_deriv(const Loop::GF3D5<const T> &gf, const Arith::simdl<T> &mask,
-               const Loop::GF3D5layout &layout,
+    calc_deriv(const Loop::GF3D2<const T> &gf, const Arith::simdl<T> &mask,
                const Arith::vect<int, Loop::dim> &I,
                const Arith::vect<T, Loop::dim> &dx) {
   using namespace Arith;
   using namespace Loop;
   // We use explicit index calculations to avoid unnecessary integer
   // multiplications
-  const T *restrict const ptr = &gf(layout, I);
+  const T *restrict const ptr = &gf(I);
   const std::array<std::ptrdiff_t, Loop::dim> offsets{
-      layout.delta(1, 0, 0),
-      layout.delta(0, 1, 0),
-      layout.delta(0, 0, 1),
+      gf.delta(1, 0, 0),
```

Comment:
I don't think these lines are correct. I don't think that the grid function class `GF3D2` has a `delta` member function.

Reply:
nice!